tactics/proof_global.ml

```diff
@@ -141,17 +141,21 @@ let private_poly_univs =
   fun () -> !b
 
 (* XXX: This is still separate from close_proof below due to drop_pt in the STM *)
-let return_proof { proof } =
-  let Proof.{name=pid;entry} = Proof.data proof in
+(* XXX: Unsafe_typ:true is needed by vio files.  *)
+let prepare_proof ~unsafe_typ { proof } =
```

Comment:
Arguments to enable hacks should probably be implicit when possible

Reply:
As this is internal I didn't worry so much, will ammend.